libs/SmartSync/SmartSyncTests/SyncManagerTests.m

```diff
@@ -763,6 +807,41 @@ - (void) testAddFilterForResync
     }
 }
 
+- (void) testResyncRunningSync
+{
+    // Create test data
+    [self createTestData];
+    
+    // Ids clause
+    NSString* idsClause = [self buildInClause:[idToNames allKeys]];
+    
+    // Create sync
+    NSString* soql = [@[@"SELECT Id, Name, LastModifiedDate FROM Account WHERE Id IN ", idsClause] componentsJoinedByString:@""];
```

Comment:
We run a slow sync down (a custom sync down that is basically a soql sync down that sleeps for a second before fetching from the server). I didn't use a semaphore here because the sync target gets serialized/deserialized and the target instance that the tests builds will not be the one used by the sync manager.

We start the sync, call re sync, expect nil, then we wait for sync and retry the resync and this time we expect it to run.